log.go

```diff
@@ -80,6 +85,76 @@ type eventLogger struct {
 	// TODO add log-level
 }
 
+type Event struct {
```

Comment:
Alright, will anyone hate me if i suggest just using a normal context with the span embedded in it on a special key, and then having a function in this package that does the `Finish` instead of having it be a method on the thing?
```
ctx := log.Start(ctx, "foobar")

log.Finish(ctx)
```

This will continue to satisfy the golang context gods: https://golang.org/src/context/context.go?s=8344:8408#L242

And doesnt provide *that* much of a worse api.

Reply:
You know, sometimes, I wonder if the go devs realize that there *are*, in fact, other people using their language...

Given that... yeah, it makes sense to just use a built-in context.

Reply:
yeah, sometimes they really make me wonder. It feels like they could just add an interface there for our custom contexts to implement